src/ProgressOnderwijsUtils/MetaObject/MetaObjectPropertyConverter.cs

```diff
@@ -39,11 +40,16 @@ struct UnusedTypeTemplate2 : IConverterSource<UnusedTypeTemplate1, int>
                 => throw new NotImplementedException();
         }
 
-        public static MetaObjectPropertyConverter GetOrNull(Type type)
-            => type
+        static readonly ConcurrentDictionary<Type, MetaObjectPropertyConverter> propertyConverterCache = new ConcurrentDictionary<Type, MetaObjectPropertyConverter>();
+
+        static readonly Func<Type, MetaObjectPropertyConverter> cachedFactoryDelegate = type =>
+            type.GetNonNullableUnderlyingType()
                 .GetInterfaces()
                 .Where(i => i.IsConstructedGenericType && i.GetGenericTypeDefinition() == typeof(IMetaObjectPropertyConvertible<,,>))
                 .Select(i => new MetaObjectPropertyConverter(i))
                 .SingleOrNull();
+
+        public static MetaObjectPropertyConverter GetOrNull(Type propertyType)

```

Comment:
...public here, so it's used not just in the MetaObjectReader.

Also, cache as early as possible, even before de-nullabization etc. because we can cache all that too.